bundles/org.openhab.binding.bluetooth.daikinmadoka/src/main/java/org/openhab/binding/bluetooth/daikinmadoka/handler/DaikinMadokaHandler.java

```diff
@@ -0,0 +1,772 @@
+                    new StringType("Off"));
+
+        }
+    }
+
+    @Override
+    public void receivedResponse(SetSetpointCommand command) {
+
+        // The update depends on the mode - so if not set - skip
+        OperationMode operationMode = this.madokaSettings.getOperationMode();
+        if (operationMode == null) {
+            return;
+        }
+
+        switch (operationMode) {
+            case HEAT:
+                this.madokaSettings.setSetpoint(command.getHeatingSetpoint());
+                break;
+            case COOL:
+                this.madokaSettings.setSetpoint(command.getCoolingSetpoint());
+                break;
+            case AUTO:
+                // Here we don't really care if we are taking cooling or heating...
+                this.madokaSettings.setSetpoint(command.getCoolingSetpoint());
+                break;
+            default:
+                return;
+        }
+
+        DecimalType dt = madokaSettings.getSetpoint();
+        if (dt != null) {
+            updateStateIfLinked(new ChannelUID(getThing().getUID(), DaikinMadokaBindingConstants.CHANNEL_ID_SETPOINT),
+                    dt);
+        }
+    }
+
+    /**
+     * Received response to "SetOperationmodeCommand" command
+     */
+    @Override
+    public void receivedResponse(SetOperationmodeCommand command) {
+        this.madokaSettings.setOperationMode(command.getOperationMode());
+        updateStateIfLinked(new ChannelUID(getThing().getUID(), DaikinMadokaBindingConstants.CHANNEL_ID_OPERATION_MODE),
+                new StringType(command.getOperationMode().toString()));
+    }
+
+    /**
+     * Received response to "SetFanSpeed" command
+     */
+    @Override
+    public void receivedResponse(SetFanspeedCommand command) {
+
+        // The update depends on the mode - so if not set - skip
+        OperationMode operationMode = this.madokaSettings.getOperationMode();
+        if (operationMode == null) {
+            return;
+        }
+
+        FanSpeed fanSpeed;
+        switch (operationMode) {
+            case HEAT:
+                fanSpeed = command.getHeatingFanSpeed();
+                this.madokaSettings.setFanspeed(fanSpeed);
+                break;
+            case COOL:
+                fanSpeed = command.getCoolingFanSpeed();
+                this.madokaSettings.setFanspeed(fanSpeed);
+                break;
+            case AUTO:
+                fanSpeed = command.getCoolingFanSpeed(); // Arbitrary cooling or heating... They are the same!
+                this.madokaSettings.setFanspeed(fanSpeed);
+                break;
+            default:
+                return;
+        }
+
+        updateStateIfLinked(new ChannelUID(getThing().getUID(), DaikinMadokaBindingConstants.CHANNEL_ID_FAN_SPEED),
+                new DecimalType(fanSpeed.value()));
+    }
+
+    private void updateStateIfLinked(ChannelUID channelUID, State state) {
+        if (isLinked(channelUID)) {
```

Comment:
This should cleanup the code a little bit
```suggestion
    private void updateStateIfLinked(String channelId, State state) {
        ChannelUID channelUID = new ChannelUID(getThing().getUID(),channelId);
        if (isLinked(channelUID)) {
```